flask_ninja/operation.py

```diff
@@ -187,179 +158,200 @@ def serialize(cls, resp: Any) -> Any:
+    def _parse_path_params(self, path: str) -> list[str]:
+        return re.findall(r"(\w+)>", path)
+
+    def _parse_params(self, path: str) -> list[ModelField]:
+        path_param_names = self._parse_path_params(path)
 
-        The idea is:
-        - if a param is in endpoint path string - it is a path param
-        - if it's not and it's a complex type - list, dict, object - it is body param
-        - otherwise it is query param
-        """
         param_docs = {
             param.arg_name: param.description or ""
             for param in doc_parse(self.view_func.__doc__ or "").params
         }
 
-        path_params = self._parse_path_params(path, param_docs)
-        query_params, body_params = self._parse_func_params(param_docs)
-
-        for param_name, param in path_params.items():
-            # We recognized the param as path param, not query param
-            if param_name in query_params:
-                if param.schema["schema"]["type"] is None:
-                    param.schema["schema"]["type"] = query_params[param_name].schema[
-                        "schema"
-                    ]["type"]
-                elif (
-                    param.schema["schema"]["type"]
-                    != query_params[param_name].schema["schema"]["type"]
-                ):
-                    raise ApiConfigError(
-                        f"Function requires {param_name} argument of type {query_params[param_name].schema['schema']['type']}, got {param.schema['schema']['type']}"
-                    )
-                del query_params[param_name]
-            elif param_name in body_params:
-                raise ApiConfigError(
-                    f"Param of type {type(body_params[param_name])} can't be path param."
-                )
-            else:
-                raise ApiConfigError(f"Function is missing {param_name} argument")
+        fields = []
 
-        if len(body_params) > 1:
-            raise ApiConfigError("Multiple complex objects in function arguments.")
+        for param in inspect.signature(self.view_func).parameters.values():
+            model_field = get_param_model_field(
+                param=param,
+                force_type=ParamType.PATH if param.name in path_param_names else None,
+            )
+            if param.name in param_docs and not model_field.field_info.description:
+                model_field.field_info.description = param_docs[param.name]
 
-        return path_params | query_params | body_params
+            fields.append(model_field)
+
+        if (
+            len(
+                list(
+                    field
+                    for field in fields
+                    if field.field_info.in_ == ParamType.BODY  # type:ignore
+                )
+            )
+            > 1
+        ):
+            raise ApiConfigError("Multiple request body arguments.")
+
+        for path_param in path_param_names:
+            if not any(
```

Comment:
from my point of view, it can be simplified a bit

```python
        handler_path_param_names = [
            param.name 
            for param in fields 
            if param.field_info.in_ == ParamType.PATH
        ]

        for path_param in path_param_names:
            if path_param not in handler_path_param_names:
                raise ApiConfigError(f"API handler misses {path_param} argument.")    

```

Reply:
improved together with the previous check